src/graph.rs

```diff
@@ -427,6 +427,41 @@ impl PyGraph {
         Ok(())
     }
 
+    /// Remove a node from the graph and add edges from all predecessors to all
+    /// descendants
+    ///
+    /// :param int node: The index of the node to remove. If the index is not
+    ///     present in the graph it will be ingored and this function willl have
+    ///     no effect.
+    #[text_signature = "(node, /)"]
+    pub fn remove_node_retain_edges(
+        &mut self,
+        py: Python,
+        node: usize,
+    ) -> PyResult<()> {
+        let index = NodeIndex::new(node);
+        let mut edge_list: Vec<(NodeIndex, NodeIndex, PyObject)> = Vec::new();
+        for (source, weight) in self
+            .graph
+            .edges_directed(index, petgraph::Direction::Incoming)
```

Comment:
I haven't looked at the graph class much, but does petgraph ignore direction if using a graph instead of a digraph?

Reply:
Yeah, if you call `edges_directed` on an undirected graph it returns all edges with the target or source set based on the direction: https://docs.rs/petgraph/0.5.1/petgraph/stable_graph/struct.StableGraph.html#method.edges_directed

But, I misread that doc and I need to tweak this so it only loops once because right now it's adding 2 edges between all neighbors and it should just be one.

Reply:
I decided to just remove the `PyGraph` implementation for now: https://github.com/Qiskit/retworkx/pull/134/commits/bb82c45649f044d88e26dbca090acabaf063b6f5

The only use case we have is for the directed graphs. If we need it for undirected graphs in the future we can add it.